src/api/auth/auth.post.test.ts

```diff
@@ -0,0 +1,47 @@
+import { after, before, describe, it } from 'node:test';
+import { prisma } from '@core/db/db.js';
+import { hash } from '@core/encryption/hash.js';
+import axios from 'axios';
+import { assert, expect } from 'chai';
+
+const USER_TO_CREATE = {
+  name: 'John Doe',
+  email: 'john.doe@example.com',
+  password: 'password123',
+  birthDate: '1990-01-01T00:00:00.000Z',
+};
+
+describe('User Authentication', () => {
+  it('should authenticate a user', async () => {
+    const response = await axios.post('http://localhost:8080/auth', {
+      email: USER_TO_CREATE.email,
+      password: USER_TO_CREATE.password,
+    });
+
+    expect(response.status).to.equal(200);
+
+    const user = response.data.user;
+
+    const dbUser = await prisma.user.findFirst({
+      where: { email: USER_TO_CREATE.email },
+    });
+
+    expect(response.data).to.have.property('token');
```

Comment:
Importante testar o body de resposta completamente com `.to.be.deep.eq`, mas isso pode ser feito quando o token tiver completo aqui.